### PR TITLE
pylightning: Fix the parameter `label` of `sendpay` 

### PR DESCRIPTION
**Pull request by trueptolemy**
`description` is renamed as `label` in 0.7.0, so now use `label` as the field name in pylightning.

### CODE REVIEW

File: contrib/pylightning/lightning/lightning.py [hunk 1]
```diff
@@ -804,18 +804,37 @@ def plugin_rescan(self):
...
 
+    def sendpay(self, route, payment_hash, *args, **kwargs):
+        """
+        Send along {route} in return for preimage of {payment_hash}
+        """
+
+        if 'description' in kwargs:
+            return self._deprecated_sendpay(route, payment_hash, *args, **kwargs)
+
+        def _sendpay(route, payment_hash, label=None, msatoshi=None):
```
Comment (ZmnSCPxj) [inline contrib/pylightning/lightning/lightning.py]: I find it strange that this function is a sub-function here, but `_deprecated_sendpay` is a method of the class.  Suggest making `_deprecated_sendpay` into a sub-function here as well.

Comment (trueptolemy) [reply contrib/pylightning/lightning/lightning.py]: Good idea! Now I make it as a sub-function. I also make similar changes in #3073  .

File: lightningd/pay.c [hunk 2]
```diff
@@ -804,17 +804,43 @@ static struct command_result *json_sendpay(struct command *cmd,
...
+		const char *desc = NULL;
+		if (!param(cmd, buffer, params,
+			   p_req("route", param_array, &routetok),
+			   p_req("payment_hash", param_sha256, &rhash),
+			   p_opt("label", param_escaped_string, &label),
+			   p_opt("description", param_escaped_string, &desc),
+			   p_opt("msatoshi", param_msat, &msat),
+			   p_opt("bolt11", param_string, &b11str),
+			   NULL))
+		return command_param_failed();
```
Comment (cdecker) [inline lightningd/pay.c]: Is the indentation of the `return`s off? They seem aligned with the `if` above them. Made me check twice how the 2 lines below aren't unreachable.

File: contrib/pylightning/lightning/lightning.py (hunk 1 above)
Comment (cdecker) [reply contrib/pylightning/lightning/lightning.py]: I think wrapping this in sub-functions is a bit strange. It's just a single `warning` and an assignment that differs.

File: contrib/pylightning/lightning/lightning.py [hunk 3]
```diff
@@ -804,17 +804,36 @@ def plugin_rescan(self):
...
         """
-        payload = {
-            "route": route,
-            "payment_hash": payment_hash,
-            "description": description,
-            "msatoshi": msatoshi,
-        }
-        return self.call("sendpay", payload)
+
+        def _deprecated_sendpay(route, payment_hash, description=None, msatoshi=None):
```
Comment (cdecker) [inline contrib/pylightning/lightning/lightning.py]: Giving `description` a default value seems strange here, after all we make sure we call it with a description below.

File: lightningd/pay.c (hunk 2 above)
Comment (trueptolemy) [reply lightningd/pay.c]: This is an alignment error, my carelessness. :-(

File: contrib/pylightning/lightning/lightning.py (hunk 3 above)
Comment (trueptolemy) [reply contrib/pylightning/lightning/lightning.py]: Yes! I should remove this default.

File: contrib/pylightning/lightning/lightning.py (hunk 1 above)
Comment (trueptolemy) [reply contrib/pylightning/lightning/lightning.py]: Yeah. Let this function be independent。

File: lightningd/pay.c (hunk 2 above)
Comment (cdecker) [reply lightningd/pay.c]: No problem, was just really confused :wink:

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/3089/commits/dbf1d25463601565a19f71f0668bce763cdd45f1

**Pull request merged**